src/plugins/english/rainofsnow.ts

```diff
@@ -0,0 +1,186 @@
+    const novel: Plugin.SourceNovel & { totalPages: number } = {
+      path: novelPath,
+      name: '',
+      totalPages: 0,
+    };
+
+    const body = await fetchApi(this.site + novelPath).then(res => res.text());
+    const loadedCheerio = parseHTML(body);
+
+    novel.name = loadedCheerio('div.text > h2').text().trim();
+
+    novel.cover = loadedCheerio('.imagboca1 > img').attr('data-src');
+
+    novel.summary = loadedCheerio('#synop').text().trim();
+
+    novel.genres = loadedCheerio(
+      'body > div.queen > div > div > div.row > div.col-md-12.col-lg-7 > div > div.backcolor1 > ul > li:nth-child(5) > small',
+    )
+      .text()
+      .trim();
+
+    novel.author = loadedCheerio(
+      'body > div.queen > div > div > div.row > div.col-md-12.col-lg-7 > div > div.backcolor1 > ul > li:nth-child(2) > small',
+    ).text();
+
+    let x = 1;
+    loadedCheerio('.page-numbers')
+      .find('li')
+      .each(function (i, el) {
+        const num = loadedCheerio(el).find('a').text().trim().match(/(\d+)/);
+
+        let n = Number(num?.[1] || '0');
+        if (n > x) {
+          x = n;
+        }
+      });
+
+    novel.totalPages = x;
+
+    novel.status = status ? NovelStatus.Completed : NovelStatus.Ongoing;
+    novel.chapters = this.parseChapters(loadedCheerio);
+    return novel;
+  }
+
+  // parse paged chapters
+  async parsePage(novelPath: string, page: string): Promise<Plugin.SourcePage> {
+    const url = this.site + novelPath + '/page/' + page + '/#chapter';
+    const body = await fetchApi(url).then(res => res.text());
+    const loadedCheerio = parseHTML(body);
+    const chapters = this.parseChapters(loadedCheerio);
+    return { chapters };
+  }
+
+  // helper to parse a novel
+  parseChapters(loadedCheerio: CheerioAPI) {
+    const chapter: Plugin.ChapterItem[] = [];
+
+    loadedCheerio('#chapter')
+      .find('li')
+      .each((i, el) => {
+        const name = loadedCheerio(el).find('.chapter').first().text().trim();
+        const releaseTime = loadedCheerio(el).find('small').text();
+        const path = loadedCheerio(el)
+          .find('a')
+          .attr('href')
+          ?.slice(this.site.length);
+
+        if (path && name) {
+          chapter.push({ name, path, releaseTime });
+        }
+      });
+
+    return chapter;
+  }
+
+  async parseChapter(chapterPath: string): Promise<string> {
+    // parse chapter text here
+    const result = await fetch(this.site + chapterPath);
+    const body = await result.text();
+
+    const loadedCheerio = parseHTML(body);
+
+    let chapterName = loadedCheerio('.content > h2').text();
+    let chapterText = loadedCheerio('.content').html();
+    if (!chapterText) return '';
+    return chapterName + '\n' + chapterText;
+  }
+
+  async searchNovels(
+    searchTerm: string,
+    pageNo: number,
+  ): Promise<Plugin.NovelItem[]> {
+    const novels: Plugin.NovelItem[] = [];
+    // get novels using the search term
+    // no page number, infinite scroll
+
```

Comment:
You should return undefined or empty array if page > 1. That is a signal for end of infinite fetching.